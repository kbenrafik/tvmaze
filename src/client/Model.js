import _ from 'lodash';

/**
 * Abstraction of model
 * @class
 */
class Model {
  /**
   * @constructor
   * @param {Object} data
   */
  constructor(data = {}) {
    this.data = data;
  }

  /**
   * Get property from the model
   * @param {String} property
   * @returns {*}
   */
  get(property) {
    return _.get(this.data, property);
  }

  /**
   * Get id
   * @returns {*}
   */
  getId() {
    return this.get('id');
  }

  /**
   * Get name.
   * @returns {*}
   */
  getName() {
    return this.get('name');
  }

  /**
   * Get summary.
   * @returns {*}
   */
  getSummary() {
    return this.get('summary');
  }

  /**
   * Get image.
   * @param {String} size of the image, by default is the original.
   * @returns {*}
   */
  getImage(size = 'original') {
    return this.get('image.original');
  }
}

export default Model;