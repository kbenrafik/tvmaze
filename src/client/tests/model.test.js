import Model from '../Model';

describe('Model', () => {
  let mockData = {};

  beforeEach(() => {
    mockData = {
      id: 45,
      name: 'is a name',
      summary: 'summary show',
      image: {
        original: '/link/original',
        medium: '/link/medium'
      },
      other: 'Ã­s other'
    };
  });
  it('should instantiate', () => {
    const model = new Model(mockData);
    expect(model).toMatchSnapshot();
  });

  it('should get work', () => {
    const model = new Model(mockData);
    expect(model.get('other')).toBe(mockData.other);
  });

  it('should getter works', () => {
    const model = new Model(mockData);
    expect(model.getId()).toBe(mockData.id);
    expect(model.getSummary()).toBe(mockData.summary);
    expect(model.getImage('medium')).toBe(mockData.image.medium);
    expect(model.getImage()).toBe(mockData.image.original);
    expect(model.getName()).toBe(mockData.name);
  });
});
