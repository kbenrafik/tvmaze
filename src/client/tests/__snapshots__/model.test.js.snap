// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model should instantiate 1`] = `
Model {
  "data": Object {
    "id": 45,
    "image": Object {
      "medium": "/link/medium",
      "original": "/link/original",
    },
    "name": "is a name",
    "other": "Ã­s other",
    "summary": "summary show",
  },
}
`;
